### DISCUSSION TITLE
Setting an arbitrary GUC via server config / env variables?

### DISCUSSION THREAD

**Discussion by EvanK** [hasura/graphql-engine#8523] [Question] [answered]
I know there are [various command flags or environment variables](https://hasura.io/docs/latest/graphql/core/deployment/graphql-engine-flags/reference/) for configuring the Hasura server itself, but is there any way to set arbitrary GUC variables that have nothing to do with Hasura?

My use case here is another service I run on top of the same Postgres database that expects and uses certain GUC values, and I'd like to be able to call some of the same functions but they will fail if said GUCs are not defined.

The values for these would be available on startup of the Hasura (and other) services, but not for every incoming request, hence why I'd want to set them on server start.

**Addendum:** Specifically, I was hoping to be able to provide an extra environment variable like `HASURA_GRAPHQL_GUC_ARBITRARY_VALUE=foo bar` and have it do the equivalent of `SET LOCAL arbitrary.value = 'foo bar';`

**Comment by BenoitRanque** [comment, marked as answer]
Hasura does not run in the same process or in the same machine as your database, so they do not share environments.
The closest you can do: set the env var as part of the session, (either via JWT or webhook auth mode), and access them in SQL with the rest of your session: https://hasura.io/docs/latest/graphql/core/guides/auditing-tables/

This only works for mutations. Queries can access the same variable when executing custom SQL functions that receive the session as an argument.

**I would strongly advise _against_ any of the above for your use case**

I understand your use case is setting these environment variables on server start, but the question is which server?
Your hasura instance does not share the lifetime of your database, it is possible for the database to restart while the hasura instance stays online.

I would encourage you to instead set these env vars in your database server, on server start.
